app/src/main/java/com/example/swugger/AddReminderDialogFragment.java

```diff
@@ -0,0 +1,118 @@
+package com.example.swugger;
+
+import android.app.Dialog;
+import android.app.DialogFragment;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.support.v7.app.AlertDialog;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Spinner;
+
+/**
+ * Created by David on 8/11/2018.
+ */
+
+public class AddReminderDialogFragment extends DialogFragment implements AdapterView.OnItemSelectedListener {
+
+    private static final int DAY_SPINNER_DEFAULT_POS = 0;
+    private static final int HOUR_SPINNER_DEFAULT_POS = 0;
+    private static final int MINUTE_SPINNER_DEFAULT_POS = 15;
+    private EditEventDialogFragment mCallback;       // should be set by caller
+    private Spinner mDaySpinner;
+    private Spinner mHourSpinner;
+    private Spinner mMinuteSpinner;
+    private ArrayAdapter daySpinnerAdapter;
+    private ArrayAdapter hourSpinnerAdapter;
+    private ArrayAdapter minuteSpinnerAdapter;
+    private Integer daysBefore;
+    private Integer hoursBefore;
+    private Integer minutesBefore;
+
+    @Override
+    public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+        switch (parent.getId()) {
+            case R.id.spinner_days_new_reminder_dialog:
+                daysBefore = Integer.parseInt((String) parent.getItemAtPosition(position));
+
+            case R.id.spinner_hours_new_reminder_dialog:
+                hoursBefore = Integer.parseInt((String) parent.getItemAtPosition(position));
+
+            case R.id.spinner_minutes_new_reminder_dialog:
+                minutesBefore = Integer.parseInt((String) parent.getItemAtPosition(position));
+        }
+    }
+
+    @Override
+    public void onNothingSelected(AdapterView<?> parent) {
+
+    }
+
+    /** MUST CALL
+     *  Attaches this DialogFragment to another DialogFragment (in this case, the edit events DialogFragment. */
```

Comment:
What's going on here? Why would you have another DialogFragment on the Screen?

Also, why are you using DialogFragment opposed to a regular Dialog?

Reply:
i've read that Google wants people to use DialogFragments for anything Dialog related now even for simple alert dialogs

Reply:
I'm just iffy about it since do you really need to create an object with its own lifecycle in it? Feels like it's just going to be taxing to the system and annoying to work with. Not 100% sure on what you should do here, but just my tidbit on why I thought a DialogFragment would be weird here.

Reply:
yeah same thats how i came across the Google statement thing, i felt like it was overkill too, esp since some of the stuff i use it for is pretty basic and could just be a simple alert dialog